src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,183 @@
+     */
+    public Parser(Storage list, Ui ui) {
+        this.storage = list;
+        this.ui = ui;
+    }
+
+    /**
+     * Handles parsing of text input by the user.
+     *
+     * @return
+     * @throws IOException
+     */
+    public static String parse(String text) throws IOException {
+
+        if (text.equalsIgnoreCase("list")) {
+            return ui.list();
+
+        } else if (text.equalsIgnoreCase("bye")) {
+            return ui.bye();
+
+        } else if (text.startsWith("delete")) {
+            try {
+                if (text.equalsIgnoreCase("delete") || text.equalsIgnoreCase("delete ")) {
+                    return ui.deleteNoNumber();
+                } else {
+                    int deletable = Integer.parseInt(text.replace("delete ", "")) - 1;
+                    if (deletable < storage.taskList.arrayList.size() && deletable >= 0) {
+                        Task deleted = storage.taskList.arrayList.get(deletable);
+                        storage.taskList.delete(deletable);
+                        return ui.delete(deleted);
+                    } else {
+                        return ui.deleteError();
+                    }
+
+                }
+            } catch (NumberFormatException error) {
+                return ui.deleteNumberFormatError();
+            }
+
+        } else if (text.startsWith("todo")) {
+            if (text.equalsIgnoreCase("todo") || text.equalsIgnoreCase("todo ") || text.replace("todo ", "").trim().length() < 1) {
+                try {
+                    throw new DukeException.DukeToDoException("Please provide a description for your todo task.");
+                } catch (DukeException.DukeToDoException error) {
+                    return ui.todoError(error);
+                }
+            } else {
+                ToDo item = new ToDo(text.replace("todo ", ""));
+                storage.taskList.addTodo(item);
+                return ui.todo(item);
+            }
+
+        } else if (text.startsWith("deadline")) {
+            try {
+                String[] description = text.replace("deadline ", "").split("/by ");
+                try {
+                    if (description[1].length() > 10) {
+                        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+                        Deadline item = new Deadline(description[0], LocalDateTime.parse(description[1], formatter));
+                        storage.taskList.addDeadline(item);
+                        return ui.deadline(item);
+                    } else {
+                        description[1] = description[1] + " 00:00";
+                        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+                        Deadline item = new Deadline(description[0], LocalDateTime.parse(description[1], formatter));
+                        storage.taskList.addDeadline(item);
+                        return ui.deadline(item);
```

Comment:
Might be neater to abstract this into another sub-function due to the repetition